File: ros2_ws/src/cpp_srvcli/src/add_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using AddTwoInts = example_interfaces::srv::AddTwoInts;

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    // 創一個名叫 "add_two_ints_client" 的 Node
    auto node = rclcpp::Node::make_shared("add_two_ints_client");
    // 創建一個 "add_two_ints" 這個 Service 的 Client 端
    auto client = node->create_client<AddTwoInts>("add_two_ints");

    while (!client->wait_for_service(std::chrono::seconds(1))) {
        RCLCPP_INFO(node->get_logger(), "Waiting for service...");
    }

    auto request = std::make_shared<AddTwoInts::Request>();
    request->a = 3;
    request->b = 7;
		
		// 發送請求給 Server 端
    auto result = client->async_send_request(request);
    // 等待 Server 端回傳資料
    if (rclcpp::spin_until_future_complete(node, result) == rclcpp::FutureReturnCode::SUCCESS) {
        RCLCPP_INFO(node->get_logger(), "Result: %ld", result.get()->sum);
    } 
    else {
        RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Failed to call service add_three_doubles");    // CHANGE
    }
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/py_srvcli/py_srvcli/add_two_ints_client.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class AddTwoIntsClient(Node):
    def __init__(self):
        super().__init__('add_two_ints_client')
        # 建立 Client 端，參數分別為：
        # (使用的interface(資料格式), Service名稱)
        self.client = self.create_client(AddTwoInts, 'add_two_ints')
        while not self.client.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('Waiting for service...')
        self.send_request()

    def send_request(self):
        req = AddTwoInts.Request()
        req.a = 3
        req.b = 5
        # 發送 request
        future = self.client.call_async(req)
        # 等待 Server 端回傳資料
        rclpy.spin_until_future_complete(self, future)
        if future.result():
            self.get_logger().info(f'Result: {future.result().sum}')

def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()
    rclpy.shutdown()

if __name__ == '__main__':
    main()